README.md

```diff
@@ -22,50 +22,108 @@ although basic usage is shown below.
+  const result = u2f.checkRegistration(req.session.registrationRequest, req.body.registrationResponse);
 
-// 4. (Server) Check registration result.
-var checkres = u2f.checkRegistration(session.authRequest, res);
+  if (result.successful) {
+    // Success!
+    // Save result.publicKey and result.keyHandle to the server-side datastore, associated with
+    // this user.
+    return res.sendStatus(200);
+  }
 
-if (checkres.successful) {
-    // Registration successful, save 
-    // checkres.keyHandle and checkres.publicKey to user's account in your db.
-} else {
-    // checkres.errorMessage will contain error text.
+  // result.errorMessage is defined with an English-language description of the error.
+  return res.send({result});
 }
 ```
 
+##### Client logic:
+
+Note that the `window.u2f` object is defined in the official [Javascript U2F API](https://github.com/google/u2f-ref-code), for which a polyfill is [available as an npm module](https://www.npmjs.com/package/u2f-api-polyfill).
+
+```javascript
+const registrationRequest = ...  // Retrieve this from hitting the registration challenge endpoint
+
+window.u2f.register(registrationRequest.appId, [registrationRequest], [], (registrationResponse) => {
+  // Send this registration response to the registration verification server endpoint
+});
+```
+
 ### User Authentication Flow
+
+##### Server endpoints:
+
 ```javascript
-// 1. Check that the user is logged in using password authentication (or some other way).
+const u2f = require('u2f');
+
+function authenticationChallengeHandler(req, res) {
+  // 1. Check that the user is logged in using password authentication.
+
+  // 2. Fetch the user's key handle from the server-side datastore. This field should have been
+  // saved after the registration procedure.
+  const keyHandle = ...
 
-// 2. (On server) generate user sign request using keyHandle from user account.
-var req = u2f.request("<appId>", user.keyHandle);
-session.authRequest = req;
+  // 3. Generate an authentication request and save it in the session. Use the same app ID that
+  // was used in registration!
+  const authRequest = u2f.request(APP_ID, keyHandle);
+  req.session.authRequest = authRequest;
 
-// 3. (On client) use U2F API to request signature.
-// u2f.sign([req], function(res) { /* send res back to server */ });
+  // 4. Send the authentication request to the client, who will use the Javascript U2F API to sign
+  // the authentication request, and send it back to the server for verification.
+  return res.send(authRequest);
+}
 
-// 4. (On server) check signature using publicKey from user account.
-var checkres = u2f.checkSignature(session.authRequest, res, user.publicKey);
+function authenticationVerificationHandler(req, res) {
+  // 4. Verify the authentication response from the client against the authentication request saved
+  // in the server-side session.
+  const result = u2f.checkSignature(req.session.authRequest, req.body.authResponse);
```

Comment:
For `checkSignature()` the third argument (user's publicKey) is missing.
I got a buffer error, after looking at the code I saw that there are three arguments required.